src/Main.elm

```diff
@@ -41,25 +147,151 @@ update msg model =
 view : Model -> Html Msg
 view model =
     div []
-        [ div [ class "book-list" ] (List.map book model.books)
+        [ div [ class "book-list" ] (List.map (\b -> bookComponent b model.cart) model.books)
+        , div [ class "cart" ] [ cartView model ]
         ]
 
 
-book : Book -> Html Msg
-book b =
+inCart : BookID -> Cart -> Bool
+inCart bookID cart =
+    List.any (\cartItem -> cartItem.id == bookID) cart
```

Comment:
I'd move this FN out from among the view FNs - it's not a view concern, belongs more with the model/update FNs.